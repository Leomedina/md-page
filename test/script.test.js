const showdown = require('../src/showdown');

describe("HTML Rendering Tests: Ensures header syntax renders accurately.", () => {
  test('It should render # as <h1>', () => {
    const testCase = "# hello";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test('It should render ## as <h2>', () => {
    const testCase = "## hello";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test('It should render ### as <h3>', () => {
    const testCase = "### hello";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test('It should render #### as <h4>', () => {
    const testCase = "#### hello";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test('It should render ##### as <h5>', () => {
    const testCase = "##### hello";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test('It should render ###### as <h6>', () => {
    const testCase = "###### hello";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });
});

describe("HTML Rendering Tests: Ensures Emphasis syntax renders accurately ", () => {
  test("It should render *text* as italics", () => {
    const testCase = "*This text will be italic*";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test("It should render _text_ as italics", () => {
    const testCase = "_This text will be italic_";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test("It should render **text** as bold", () => {
    const testCase = "**This text will be bold**";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test("It should render __text__ as bold", () => {
    const testCase = "__This text will be bold__";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test("It should render a combined bold and italics string __text **text**__", () => {
    const testCase = "_You **can** combine them_";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

});

describe("HTML Rendering Tests: Ensures Lists syntax renders accurately", () => {
  test("It should render unordered lists of a single item", () => {
    const testCase = "* Item 1";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test("It should render unordered lists of multiple items", () => {
    const testCase = "* Item 1\n * Item 2\n * Item 3";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test("It should render unordered lists of multiple items with indented items", () => {
    const testCase = "* Item 1\n * Item 2\n * Item 3 \n\t * Item 4\n\t * Item 5";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });
});

describe("HTML Rendering Tests: Ensures Images and Links are rendered accurately", () => {
  test("It should render images", () => {
    const testCase = "![TestPassed](https://media0.giphy.com/media/111ebonMs90YLu/giphy.gif)";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test("It should render links", () => {
    const testCase = "[md-page](https://github.com/oscarmorrison/md-page)"
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });
});

describe("HTML Rendering Tests: Ensures Blockquotes are rendered accurately", () => {
  test("It should render a single blockquote", () => {
    const testCase = ">This should be a blockquote"
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });

  test("It should render multiple-line blockquotes", () => {
    const testCase = ">Line 1\nLine 2\nLine 3";
    expect(new showdown.Converter().makeHtml(testCase)).toMatchSnapshot();
  });
});