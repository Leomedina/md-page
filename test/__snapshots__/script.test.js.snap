// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML Rendering Tests: Ensures Blockquotes are rendered accurately It should render a single blockquote 1`] = `
"<blockquote>
  <p>This should be a blockquote</p>
</blockquote>"
`;

exports[`HTML Rendering Tests: Ensures Blockquotes are rendered accurately It should render multiple-line blockquotes 1`] = `
"<blockquote>
  <p>Line 1
  Line 2
  Line 3</p>
</blockquote>"
`;

exports[`HTML Rendering Tests: Ensures Emphasis syntax renders accurately  It should render **text** as bold 1`] = `"<p><strong>This text will be bold</strong></p>"`;

exports[`HTML Rendering Tests: Ensures Emphasis syntax renders accurately  It should render *text* as italics 1`] = `"<p><em>This text will be italic</em></p>"`;

exports[`HTML Rendering Tests: Ensures Emphasis syntax renders accurately  It should render __text__ as bold 1`] = `"<p><strong>This text will be bold</strong></p>"`;

exports[`HTML Rendering Tests: Ensures Emphasis syntax renders accurately  It should render _text_ as italics 1`] = `"<p><em>This text will be italic</em></p>"`;

exports[`HTML Rendering Tests: Ensures Emphasis syntax renders accurately  It should render a combined bold and italics string __text **text**__ 1`] = `"<p><em>You <strong>can</strong> combine them</em></p>"`;

exports[`HTML Rendering Tests: Ensures Images and Links are rendered accurately It should render images 1`] = `"<p><img src=\\"https://media0.giphy.com/media/111ebonMs90YLu/giphy.gif\\" alt=\\"TestPassed\\" /></p>"`;

exports[`HTML Rendering Tests: Ensures Images and Links are rendered accurately It should render links 1`] = `"<p><a href=\\"https://github.com/oscarmorrison/md-page\\">md-page</a></p>"`;

exports[`HTML Rendering Tests: Ensures Lists syntax renders accurately It should render unordered lists of a single item 1`] = `
"<ul>
<li>Item 1</li>
</ul>"
`;

exports[`HTML Rendering Tests: Ensures Lists syntax renders accurately It should render unordered lists of multiple items 1`] = `
"<ul>
<li>Item 1</li>
<li>Item 2</li>
<li>Item 3</li>
</ul>"
`;

exports[`HTML Rendering Tests: Ensures Lists syntax renders accurately It should render unordered lists of multiple items with indented items 1`] = `
"<ul>
<li>Item 1</li>
<li>Item 2</li>
<li>Item 3 <ul>
<li>Item 4</li>
<li>Item 5</li></ul></li>
</ul>"
`;

exports[`HTML Rendering Tests: Ensures header syntax renders accurately. It should render # as <h1> 1`] = `"<h1 id=\\"hello\\">hello</h1>"`;

exports[`HTML Rendering Tests: Ensures header syntax renders accurately. It should render ## as <h2> 1`] = `"<h2 id=\\"hello\\">hello</h2>"`;

exports[`HTML Rendering Tests: Ensures header syntax renders accurately. It should render ### as <h3> 1`] = `"<h3 id=\\"hello\\">hello</h3>"`;

exports[`HTML Rendering Tests: Ensures header syntax renders accurately. It should render #### as <h4> 1`] = `"<h4 id=\\"hello\\">hello</h4>"`;

exports[`HTML Rendering Tests: Ensures header syntax renders accurately. It should render ##### as <h5> 1`] = `"<h5 id=\\"hello\\">hello</h5>"`;

exports[`HTML Rendering Tests: Ensures header syntax renders accurately. It should render ###### as <h6> 1`] = `"<h6 id=\\"hello\\">hello</h6>"`;
